crates/wysiwyg/src/dom/range.rs

```diff
@@ -38,3 +39,19 @@ pub struct SameNodeRange {
     /// The position within this node that corresponds to the end of the range
     pub end_offset: usize,
 }
+
+impl Display for Range {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Range::SameNode(range) => f.write_str(
+                format!(
+                    "SameNode: {} {}",
+                    range.start_offset, range.end_offset
+                )
+                .as_str(),
+            ),
+            Range::TooDifficultForMe => f.write_str("TooDifficultForMe"),
+            Range::NoNode => f.write_str("NoNode"),
+        }
+    }
+}
```

Comment:
Since Range already implements Debug, you can just use `{:?}` in the formatting code inside `replace_text_in`, and then you won't need to impl `Display` here.